packages/grafana-ui/src/components/Segment/useExpandableLabel.tsx

```diff
@@ -1,10 +1,18 @@
 import React, { useState, useRef, ReactElement } from 'react';
 
-export const useExpandableLabel = (initialExpanded: boolean) => {
+interface LabelProps {
+  Component: ReactElement;
+  onClick?: () => void;
+}
+
+export const useExpandableLabel = (
+  initialExpanded: boolean
+): [React.ComponentType<LabelProps>, number, boolean, (expanded: boolean) => void] => {
   const ref = useRef<HTMLDivElement>(null);
-  const [expanded, setExpanded] = useState(initialExpanded);
-  const [width, setWidth] = useState();
-  const Label = ({ Component, onClick }: { Component: ReactElement; onClick: () => void }) => (
+  const [expanded, setExpanded] = useState<boolean>(initialExpanded);
+  const [width, setWidth] = useState<number>(0);
```

Comment:
Doesn't TS infer the types from the initial state args?

Reply:
It does, added type first, then initial value:) good catch